plugin/fittencode.vim

```diff
@@ -163,8 +163,11 @@ function! CodeCompletion()
     let b:fitten_suggestion = l:generated_text
 endfunction
 
-function! Accept()
-    echo "Accept"
+let g:fitten_accept_no_echo = 0
+function! FittenAcceptMain()
+    if g:fitten_accept_no_echo == 0
+        echo "Accept"
+    endif
     let default = pumvisible() ? "\<C-N>" : "\t"
```

Comment:
We need different `default` values based on the different mapping keys. When there are no suggestions, `default` will insert the document. Please refer to the code on line 174.

Reply:
@luozhiya

Reply:
> original keyboard input as default

We need to consider the special circumstances of all keys

> pum is an additional functionality

`pum` is necessary when we map to a `tab`

IMO, by providing two basic APIs, `FittenAccept` and `FittenAcceptable`, the interface can be more flexible, and the internal implementation can also be simplified. What's your opinion about this?

```vim
function! FittenAccept()
    echo "Accept"

    let l:text = b:fitten_suggestion

    call ClearCompletion()

    return l:text
endfunction

function! FittenAcceptable()
    return (mode() !~# '^[iR]' || !exists('b:fitten_suggestion')) ? 0 : 1
endfunction

function! FittenAcceptTab()
    if FittenAcceptable()
        return FittenAccept()
    else
        return pumvisible() ? "\<C-N>" : "\t"
    endif
endfunction

function! MapTab()
    inoremap <script><silent><nowait><expr> <Tab> FittenAcceptTab()
endfunction
```